src/mapping/Mapping.hpp

```diff
@@ -168,8 +168,14 @@ class Mapping {
    */
   void map(const time::Sample &input, Eigen::VectorXd &output, Eigen::VectorXd &initialGuess);
 
-  /// Method used by partition. Tags vertices that could be owned by this rank.
-  virtual void tagMeshFirstRound() = 0;
+  /**
+   * @brief Method used by partition. Tags vertices that could be owned by this rank.
+   *
+   * @note This function behaves differently when using single-level initialization or two-level
+   * initiliazation: in 1LI, we look at the complete mesh, whereas in 2LI, we only look at a fraction
+   * of the mesh.
```

Comment:
This is exclusively needed to raise a warning in the `tagMeshFirstRound()` in the PUM, when a 2LI is used. The information is stored in the `m2n` config and I couldn't find it anywhere else. I though a bit about itchanging the signature and concluded that it is useful, as the argument also affects the behavior of the function. This _is_ a merge conflict with #1675 (but easy to resolve).

 > also in 1LI, one might look at a fraction of the mesh if we do a bounding box filter first

That's the case if we filter on the primary rank instead of the secondary rank, right? Or are there any other configuration scenarios?

Reply:
Well, that's not what happens in practice, due to this lines I was never sure about

https://github.com/precice/precice/blob/7fc70b7916da58746c87a24c3db6ff1c171b18ce/src/precice/config/ParticipantConfiguration.cpp#L447-L451

Maybe a nice opportunity to raise the TODO question again.

Reply:
From a practical point of view, I would guess this was a simple workaround to handle globally supported RBFs (6 years ago). However,

- this makes no sense for PUM in any RBF configuration
- this makes no sense for compactly-supported RBFs for global RBFs

Reply:
Instead of adding an unused `bool usesTwoLevelInitialization` in each function, does it make sense that we give it a default value (probably false)?

Reply:
@davidscn Good observation. This should probably also give quite a speed-up for the 1LI PUM initialization :see_no_evil:

What does `confMapping.requiresBasisFunction` mean exactly? Any RBF mapping? For global basis functions in a global RBF mapping, it makes sense to switch off the filter. But not for any other case.